### PR TITLE
Add sequencer presets - Iss9

### PR DESCRIPTION
**Pull request by xavierfav**
- It is now possible to store presets on the server (in json files) #9 #44.
Some commands for saving and loading the presets are enable for admin user.
A query parameter (admin=_password_) has to be added to the url to have access to it. The password is defined in an environment variable `ADMIN_PASSWORD` (by default it is set to _admin_).

- The sequencer state is stored in an object in the client. This allows to remove some DOM access for selecting the activated pads.

- The secret session key is now hidden and can be defined with an environment variable `SESSION_SECRET`. A default value is assigned for development.

- Tthe Freesound token is now hidden and can be defined with an environment variable  `FREESOUND_TOKEN`. However I let a token in the code as a default value for development. For production, we will use another token that has unlimited Freesound access.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -174,8 +181,24 @@ io.sockets.on('connection', function (socket) {
       sequencerStates[room].gains[message[0]] = message[1];
       socket.in(room).broadcast.emit('sendGain', message);
     });
+    
+    // SAVE PRESET
+    socket.on('savePreset', function (message) {
+      console.log('recieve save preset');
```
Comment (DerouineauNicolas) [inline server.js]: Typo

File: server.js [hunk 2]
```diff
@@ -174,8 +181,24 @@ io.sockets.on('connection', function (socket) {
...
+    
+    socket.on('loadPreset', function (message) {
+      console.log('recieve load preset');
+      var preset = getPreset(message);
+      sequencerStates[room] = preset;
+      io.sockets.in(room).emit('sendSequencerPreset', JSON.stringify(preset));
+    });
 
-
+
```
Comment (DerouineauNicolas) [inline server.js]: Blank line.

**Review by DerouineauNicolas — APPROVED** [review]

File: server.js [hunk 3]
```diff
@@ -12,7 +12,10 @@ var eventEmitter = require('events').EventEmitter
 var hostname = process.env.MULT_WEB_SEQ_SERV || 'localhost';
 var base_path = process.env.BASE_PATH || '';
 var hostnamePort = process.env.MULT_WEB_SEQ_SERV_P || '8080';
+var adminPassword = process.env.ADMIN_PASSWORD || 'admin';
```
Comment (DerouineauNicolas) [inline server.js:15]: I think we should avoid the default admin password. Maybe throw an error if no environment variable is defined ?

Comment (xavierfav) [reply server.js:15]: This is just for development.
In production you should make sure you add the environment variable.

**Pull request merged**